src/com/company/Book.java

```diff
@@ -26,4 +28,22 @@ public int getPublicationYear() {
     public void setPublicationYear(int publicationYear) {
         this.publicationYear = publicationYear;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Book book = (Book) o;
+        return nameOfBook.equalsIgnoreCase(book.nameOfBook) && author.equals(book.author);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(nameOfBook, author);
+    }
+
+    @Override
+    public String toString() {
+        return "Книга: " + nameOfBook + ", " + author.toString() + ", " + publicationYear;
```

Comment:
можно опустить toString(), он будет автоматически вызван, если любой тип данных складывается со строкой
```suggestion
        return "Книга: " + nameOfBook + ", " + author + ", " + publicationYear;
```